### PR TITLE
Project restructurisation

### PR DESCRIPTION
**Pull request by stankudrow**
Changes:

- Project structure
  1. A request to drop the support for Python3.6;
  2. The pyproject.toml file contains more metadata: urls, keywords, classifiers;
  3. The project development dependencies are updated, new ones added: autopep, mypy, pytest plug-ins and so forth;
  4. The coverage has to be not less than 90% - specified in tox.ini.
  5. The black linter is removed since its formatting may violate PEP rules and make flake8 unhappy;
  6. The aiofiles directory instead of src/aiofiles does not complicate things;
  7. pre-commit is added and runs tox locally for python files to commit;
  8. pytest.ini holds necessary options, asyncio_mode is suppressed by specifying dependencies in tox.ini.

- Project functionality

1. The `wrap` function is renamed to `asyncify` and moved to aiofiles/base.py file.
2. The `isasync` function is added to identify async objects -> may be useful in `asyncify` and `if isasync(foo)`, then `return foo` - no need to run in executor an already asynchronous object - tested;
3. get_event_loop is replaced with get_running_loop + contextvars are packed into asyncified functions.

In short, more "devops", less new functionality added. Tested in all environments except Python3.6 and PyPy3.

ToDo:

- add pylint and sanitise files according to its recommendations;
- add more automatisation/devops;
- elaborate a proper git workflow if needed;
- add more documentation and build it with Sphinx;
- fixing issues would be great.

### CODE REVIEW

**Comment by Tinche** [comment]
Hi!

I would love some help maintaining this project, so a lot of these changes would be very welcome. A few points though:
* I want to keep Black
* I want to keep the src/ layout

These choices are better than the alternatives, and consistent with every other project of mine.

The rest all look good!

**Comment by stankudrow** [comment]
All right, so be src layout (`src/aiofiles/*` or `src**`?), black linter will be back. The main question is can we drop out the support for Python3.6? Its lifetime ended at the end of last year - [Stop using Python3.6](https://pythonspeed.com/articles/stop-using-python-3.6/). That means we don't need py3.6 environment in tox.ini and CI/CD pipelines anymore. Sounds good?

**Comment by Tinche** [comment]
The code should be in `src/aiofiles/*`. Here's a good article: https://hynek.me/articles/testing-packaging/

And yeah, let's drop 3.6. Users on that version can use the old versions.

Thanks!

**Comment by stankudrow** [comment]
All right. The src layout is restored, black is back and no conflicts with flake8 for now, tox works fine so far. I think this project needs documentation - sphinx or any other tool (e. g. mkdocs) can help out, I shall find a solution. If there is no other suggestions, a new PR may be ready this Sunday.
